dialects/core/normalizers.cpp

```diff
@@ -576,11 +576,11 @@ const Def* normalize_trait(const Def*, const Def* callee, const Def* type, const
         return world.lit_nat(8); // Gets lowered to function ptr
     } else if (auto size = Idx::size(type)) {
         if (size->isa<Top>()) return world.lit_nat(8);
-        if (auto w = isa_lit(size)) {
-            if (*w == 0) return world.lit_nat(8);
-            if (*w <= 0x0000'0000'0000'0100_u64) return world.lit_nat(1);
-            if (*w <= 0x0000'0000'0001'0000_u64) return world.lit_nat(2);
-            if (*w <= 0x0000'0001'0000'0000_u64) return world.lit_nat(4);
+        if (auto s = isa_lit(size)) {
+            if (*s == 0) return world.lit_nat(8); // i64
+            if (*s <= 0x0000'0000'0000'0100_u64) return world.lit_nat(1);
+            if (*s <= 0x0000'0000'0001'0000_u64) return world.lit_nat(2);
+            if (*s <= 0x0000'0001'0000'0000_u64) return world.lit_nat(4);
```

Comment:
code duplication with  size2llbits -> just use the function and divide by 8.
Or more generally, refactor the functionality out.

Reply:
yes, I stumbled over this myself. Problem is that different places are doing slightly different things. Here we are calculating the size in bytes. In the llvm backend we are mapping to llvm types - which is slightly different due to the existence of `i1`

Reply:
I could have a methods which does this and in the llvm backend a function that wraps this and puts an extra case for `i1` in front of it.

Reply:
I think the common logic should be unified or at least grouped together.
Otherwise, we have very similar (at first glance not all differences are apparent) logic all over the place that depend on each other semantically.